x-pack/libbeat/management/managerV2.go

```diff
@@ -642,15 +693,62 @@ func (cm *BeatV2Manager) reloadInputs(inputUnits []*client.Unit) error {
 		inputBeatCfgs = append(inputBeatCfgs, inputCfg...)
 	}
 
-	if !didChange(cm.lastInputCfgs, inputCfgs) {
+	if !didChange(cm.lastInputCfgs, inputCfgs) && !cm.forceReload {
 		cm.logger.Debug("Skipped reloading input units; configuration didn't change")
 		return nil
 	}
 
-	err := obj.Reload(inputBeatCfgs)
-	if err != nil {
-		return fmt.Errorf("failed to reloading inputs: %w", err)
+	if cm.forceReload {
+		cm.logger.Info("Reloading Beats inputs because forceReload is true. " +
+			"Set log level to debug to get more information about which " +
+			"inputs are causing this.")
+	}
+
+	if err := obj.Reload(inputBeatCfgs); err != nil {
+		merror := &multierror.MultiError{}
+		realErrors := multierror.Errors{}
+
+		// At the moment this logic is tightly bound to the current RunnerList
+		// implementation from libbeat/cfgfile/list.go and Input.loadStates from
+		// filebeat/input/log/input.go.
+		// If they change the way they report errors, this will break.
+		// TODO (Tiago): update all layers to use the most recent features from
+		// the standard library errors package.
+		if errors.As(err, &merror) {
+			for _, err := range merror.Errors {
+				causeErr := cause(err)
+				// A Log input is only marked as finished when all events it
+				// produceds are acked by the acker so when we see this error,
+				// we just retry until the new input can be started.
+				// This is the same logic used by the standalone configuration file
+				// reloader implemented on libbeat/cfgfile/reload.go
+				inputNotFinishedErr := &common.ErrInputNotFinished{}
+				if ok := errors.As(causeErr, &inputNotFinishedErr); ok {
```

Comment:
Consider reporting the affected unit as CONFIGURING instead of HEALTHY while this happens. We'll be able to see the state transitions in the agent logs, and CONFIGURING will map back Healthy in the Fleet UI.

Reply:
You can actually report the configuring state with a message, so you could include the files we are waiting on in that message. This will go directly to Fleet.

Reply:
I suppose the issue you'll have with this is that you can't map the error back to a unit ID. We should probably fix this. I would file a followup issue to do this.

Reply:
Agreed, not a blocker we can iterate on this in future PRs.